docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,387 @@
+    model_dir: str,
+    model_name: str,
+    target_shape: ty.Tuple[int, int, int],
+) -> None:
+    # Convert the model to tflite, with batch size 1
+    # so the graph does not have dynamic-sized tensors.
+    input = tf.keras.Input(target_shape, batch_size=1, dtype=tf.uint8)
+    output = model(input, training=False)
+    wrapped_model = tf.keras.Model(inputs=input, outputs=output)
+    converter = tf.lite.TFLiteConverter.from_keras_model(wrapped_model)
+    converter.target_spec.supported_ops = TFLITE_OPS
+    tflite_model = converter.convert()
+
+    filename = os.path.join(model_dir, f"{model_name}.tflite")
+    with open(filename, "wb") as f:
+        f.write(tflite_model)
+```
+
+When you submit a training job with this training script, this function saves the model outputs to the `model_output_directory` in the cloud.
+Once the training job is complete, Viam looks at that directory and creates a package with all of the contents of the directory, creating or updating a registry item for the ML model.
+
+{{% /expand%}}
+
+### Package the training script as a <file>tar.gz</file> source distribution
+
+You must save your training script in the `tar.gz` format to run in the Viam ML training service.
+Follow the instructions to [create a <file>tar.gz</file> gzip'd tar file](https://docs.python.org/3.10/distutils/sourcedist) from your project.
+
+{{% alert title="Tip" color="tip" %}}
+You can reference the directory structure of this [example classification training script](https://app.viam.com/packages/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb/custom-training-classification/ml_training/latest/e76d1b3b-0468-4efd-bb7f-fb1d2b352fcb).
+You must log in to the Viam app to download the package.
+Unzip the package and see <file>model/training.py</file> for an example entrypoint file.
+{{% /alert %}}
+
+## Upload a new training script or new version
+
+You must use the [Viam CLI](/cli/) to upload your training script to the Registry.
+Follow the instructions in the [CLI documentation](/cli/) to install the CLI.
+
+You can use the [`viam training-script upload`](/cli/#training-script) command to upload a new script.
+
+For example:
+
+```sh {class="line-numbers linkable-line-numbers"}
+viam training-script upload --path=<path-to-tar.gz> --org-id=<INSERT ORG ID> --script-name="MyCustomTrainingScript"
+```
+
+To find your organization's ID, navigate to your organization's **Settings** page in [the Viam app](https://app.viam.com/).
+Find **Organization ID** and click the copy icon.
+
+Once uploaded, you can view the script by navigating to the [registry's **Training Scripts** page](https://app.viam.com/registry?type=Training+Script).
+
+You can also simultaneously upload a training script and submit a training job with the [`viam train submit custom from-upload` command](/cli/#position-arguments-submit-custom).
+
+## Submit a training job
+
+You can use the Viam CLI's [`viam train submit`](/cli/#positional-arguments-submit) command to submit a training job.
```

Comment:
After uploading the training script, you can run it by submitting a training job.